azext_iot/central/models/devicetwin.py

```diff
@@ -0,0 +1,110 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import re
+from datetime import datetime, timezone, timezone
+from azext_iot.central.models.template import Template
+
+
+class DeviceTwin:
+    def __init__(
+        self, device_twin: dict,
+    ):
+        self.device_twin = device_twin
+        self.device_id = device_twin.get("deviceId")
+        self.desired_property = Property(
+            "desired property",
+            device_twin.get("properties").get("desired"),
+            self.device_id,
+        )
+        self.reported_property = Property(
+            "reported property",
+            device_twin.get("properties").get("reported"),
+            self.device_id,
+        )
+
+
+class Property:
+    def __init__(
+        self, name: str, property_collection: dict, device_id,
+    ):
+        self.name = name
+        self.property_collection = property_collection
+        self.property_collection_metadata = property_collection.get("$metadata")
+        self.capabilities_properties = self._get_capabilities(property_collection)
+        self.version = property_collection.get("$version")
+        self.device_id = device_id
+        pass
+
+    def _get_capabilities(self, d):
+        keys_to_remove = {"$metadata", "$version"}
+        return {x: d[x] for x in d if x not in keys_to_remove}
```

Comment:
do this instead:

```
{key: value for key, value in d.iteritems() if key not in keys_to_remove}
```

Please don't use single letter variables unless its a counter